server/methods/browseChannels.js

```diff
@@ -44,15 +44,24 @@ const sortUsers = function(field, direction) {
 	}
 };
 
-const getChannels = (user, canViewAnon, searchTerm, sort, pagination) => {
+const getChannelsAndGroups = (user, canViewAnon, searchTerm, sort, pagination) => {
 	if ((!user && !canViewAnon) || (user && !hasPermission(user._id, 'view-c-room'))) {
 		return;
 	}
 
 	const teams = Promise.await(Team.getAllPublicTeams());
 	const teamIds = teams.map(({ _id }) => _id);
+	const teamsMains = Rooms.findTeamMainRooms({
```

Comment:
@KevLehman 's suggestion was great, but finding and storing all main rooms here even though none of them might be used if "result" doesn't have any team doesn't seem the best approach.

My suggestion would be:
- `filter` results to get all rooms with teams
- perform a find grabbing only the name of that rooms
- `map` results to add the `belongsTo` field

rough example
```javascript
const result = Rooms.findByNameOrFNameAndRoomIdsIncludingTeamRooms( ..... );
const results = result.fetch();

const teamIds = results.filter(({ teamId }) => teamId).map(({ teamId }) => teamId);

const teams = Rooms.findByIds(teamIds, { fields: { name: 1 } });

return results.map((room) => {
  if (room.teamId) {
    const team = teams.find((team) => team._id === room.teamId);
    if (team) room.belongsTo = team.name;
  }
});